### DISCUSSION TITLE
wrong protocol in connection

### DISCUSSION THREAD

**Discussion by cordjr** [enisdenjo/graphql-ws#491] [Q&A] [answered]
**Screenshot**

![image](https://github.com/enisdenjo/graphql-ws/assets/21906/2379f3b3-97b2-4e09-adf7-b974aba80270)
![image](https://github.com/enisdenjo/graphql-ws/assets/21906/a06587e5-2b72-4c6a-9c2f-7469481266cc)

**Expected Behaviour**
Should'n t the protocol be graphql-ws instead of graphql-transport-ws ?

**Comment by enisdenjo** [comment, marked as answer]
The WebSocket subprotocol should be `graphql-transport-ws`.

**Comment by cordjr** [comment]
I must be missing something,  There are a disclaimer in your readme saying that the lib is not compatible with graphql-transport-ws .
Can you clarify this?

**Reply by enisdenjo** [reply]
[Disclaimer says the library is not backwards compatible with **subscriptions**-transport-ws](https://github.com/enisdenjo/graphql-ws#disclaimer) (not graphql-transport-ws).

Small fact: the original name of this library was graphql-transport-ws, before being renamed to just graphql-ws.

**Comment by cordjr** [comment]
Thanks for the answear

**Comment by orekav** [comment]
I am trying to connect this to AWS Appsync and if I go to this line
```js
const socket = new WebSocketImpl(typeof url === 'function' ? await url() : url, common_1.GRAPHQL_TRANSPORT_WS_PROTOCOL);
```

and change it for this, it starts working for me
```js
const socket = new WebSocketImpl(typeof url === 'function' ? await url() : url, "graphql-ws");
```

**Reply by enisdenjo** [reply]
That's because AWS AppSync supports only the [deprecated `subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws).

Also see https://github.com/enisdenjo/graphql-ws/issues/241.
